mangaki/mangaki/admin.py

```diff
@@ -229,60 +230,99 @@ def make_nsfw(self, request, queryset):
 
     make_nsfw.short_description = "Rendre NSFW les œuvres sélectionnées"
 
-    # FIXME : https://github.com/mangaki/mangaki/issues/205
     def update_tags_via_anidb(self, request, queryset):
-        if request.POST.get("post"):
-            kept_ids = request.POST.getlist('checks')
-            for anime_id in kept_ids:
-                anime = Work.objects.get(id=anime_id)
+        works = queryset.all()
 
-                retrieved_tags = anime.retrieve_tags(client)
-                deleted_tags = retrieved_tags["deleted_tags"]
-                added_tags = retrieved_tags["added_tags"]
-                updated_tags = retrieved_tags["updated_tags"]
+        if request.POST.get('confirm'): # Updating tags has been confirmed
+            to_update_work_ids = set(map(int, request.POST.getlist('to_update_work_ids')))
+            nb_updates = len(to_update_work_ids)
 
-                tags = deleted_tags
-                tags.update(added_tags)
-                tags.update(updated_tags)
+            work_ids = list(map(int, request.POST.getlist('work_ids')))
 
-                client.update_tags(anime, tags)
+            tag_titles = request.POST.getlist('tag_titles')
+            tag_weights = list(map(int, request.POST.getlist('weights')))
+            tag_anidb_tag_ids = list(map(int, request.POST.getlist('anidb_tag_ids')))
+            tags = list(map(AniDBTag, tag_titles, tag_weights, tag_anidb_tag_ids))
 
-            self.message_user(request, "Modifications sur les tags faites")
-            return None
+            # Checkboxes to know which tags have to be kept regardless of their pending status
+            tag_checkboxes = request.POST.getlist('tag_checkboxes')
+            tags_to_process = [tuple(map(int, tag_checkbox.split(':'))) for tag_checkbox in tag_checkboxes]
```

Comment:
If I'm not mistaken, since tuples are hashable, I only need to store those tuples into a set rather than a list, and keep everything else as is ? I think it would be easier than creating a dict, since I don't use the associated values but only the keys.